ec/ecresource/deploymentresource/deployment/v2/schema.go

```diff
@@ -130,6 +130,10 @@ func DeploymentSchema() schema.Schema {
 				Description: "Explicitly resets the elasticsearch_password when true",
 				Optional:    true,
 			},
+			"migrate_to_latest_hardware": schema.BoolAttribute{
+				Description: "When true, updates deployment according to the latest deployment template values.",
```

Comment:
Shall we mention that if either IC or its version is set in config, migration doesn't happen even if the flag is set?

Reply:
DOne in https://github.com/elastic/terraform-provider-ec/pull/755/commits/3af275c83d212d4d0094711db9109095e53fc7eb.